client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -205,9 +205,9 @@ private void verifyStartAndEndStreamCuts(final StreamCut startStreamCut, final S
                                 "Segment offset in startStreamCut should be <= segment offset in endStreamCut."));
 
            val fromSCSummary = startPositions.keySet()
-                                             .stream().collect(summarizingInt(Segment::getSegmentNumber));
+                                             .stream().collect(summarizingLong(Segment::getSegmentNumber));
```

Comment:
How does this method work now? It looks like it was broken before, but now it's totally wrong. If the primaryId is used for routing and the secondary is used to resolve ambiguity (And is higher order). And commits of TXNs can go in any order how do we compare to see which segment is further in the stream?

(I think the method was already broken if there was a min and a max segment that were correct and ordered. But some in the middle numerically were not)

Reply:
This was broken before and its still broken. But the degree of "brokenness" does not change. 
This is because is this method is only a partial check which checks if `from` has no segment higher and highest segment number in `to` and `to` has no segments that are lower than `lowest` segment number in `from`. 

And this is because new segment ids follow the same property - `a higher segmentId is created after a lower segmentid (since epoch is encoded as msb in segmentid so higher segment id implies higher epoch hence later creation).`

So the logic here is same as before and nothing changes from that respect.

Reply:
I created https://github.com/pravega/pravega/issues/2668